src/tests/ftest/rebuild/read_array.py

```diff
@@ -0,0 +1,141 @@
+        data_size = self.params.get("data_size", "/run/container/*")
+        obj_class = self.params.get("obj_class", "/run/container/*")
+        rank = self.params.get("rank", "/run/test/*")
+
+        server_qty = len(self.hostlist_servers)
+        replica_qty = obj_class - 14
+        total_records = object_qty * record_qty * data_size
+
+        # Do not execute any test variants that don't make sense
+        if rank > server_qty - 1:
+            self.cancel("Canceling test that would kill a non-existent rank")
+        if pool_svcn > server_qty:
+            self.cancel(
+                "Canceling test with a svcn greater than the server quantity")
+
+        # Cancel any tests with tickets already assigned
+        if server_qty < replica_qty:
+            self.cancelForTicket("DAOS-2410")
+        if rank in [1, 2]:
+            self.cancelForTicket("DAOS-2434")
+
+        # Create and connect to a pool
+        self.log.info("Creating a pool")
+        self.pool = get_pool(
+            self.context, pool_mode, pool_size, pool_name, pool_svcn,
+            self.d_log)
+
+        # Create and open a container
+        self.log.info("Creating a container")
+        self.container = get_container(self.context, self.pool, self.d_log)
+
+        # Populate the container with data on the rank to be killed
+        self.log.info("Writing array objects")
+        try:
+            written_objects = write_array_objects(
+                self.container, object_qty, record_qty, akey_size, dkey_size,
+                data_size, rank, obj_class, self.log)
+        except DaosTestError as error:
+            self.fail(
+                "Detected exception while writing data prior to "
+                "rebuild: {}".format(str(error)))
+
+        # Log rebuild status prior to killing the server
+        self.log.info("Checking pool status prior to rebuild")
+        get_pool_status(self.pool, self.log)
+
+        # Kill the server
+        self.log.info(
+            "Killing DAOS server %s (rank %s)", self.server_group, rank)
+        kill_server(
+            self.server_group, self.context, rank, self.pool, self.d_log)
+
+        # Wait for recovery to start
+        wait_for_rebuild(self.pool, self.log, True, 1)
+
+        # Read the objects during rebuild
+        message = "Reading the array objects during rebuild"
+        self.log.info(message)
+        self.d_log.info(message)
+        try:
+            read_during_rebuild(
+                self.container, self.pool, self.log, written_objects,
+                data_size, read_array_objects)
+        except DaosTestError as error:
+            # if pool_svcn
+            self.log.warning(
+                "Detected exception while reading during rebuild: %s",
+                str(error))
+
+        # Confirm rebuild completes
+        wait_for_rebuild(self.pool, self.log, False, 1)
+        self.log.info("Rebuild completion detected")
+
```

Comment:
Maybe you should verify every replica of the array object after rebuild. Ah, we do not have such api for python yet?

Reply:
I'll add a story to update this test when the API is available